celery/app/base.py

```diff
@@ -240,7 +240,12 @@ def __init__(self, main=None, loader=None, backend=None,
         self.loader_cls = loader or self._get_default_loader()
         self.log_cls = log or self.log_cls
         self.control_cls = control or self.control_cls
-        self._custom_task_cls_used = bool(task_cls)
+        self._custom_task_cls_used = (
```

Comment:
What about dropping the whole attribute here, and instead doing the following comparison in the fixup code:

```python
from celery.app import Celery
if self.task_cls == Celery.task_cls:
    <task cls wasn't overriden, proceed with task override>
```
?

This should, I think, significantly simplify the code.

Reply:
What if the user wants to override Celery to not use the DjangoTask, but the default Calary task?

Reply:
There is no functional difference between `Task` and `DjangoTask` (DT is a strict functional superset of T), so the only cases where this matters are subclass/instance checks which already sound shady and trying to support these sounds like a can of worms, hm? 

If they _really_ want `Task` and not `DT`, then maybe they are also considering disabling the whole django fixup.

Reply:
Yes it _should_ be fine, and I'm probably being extra careful. 

I'm trying to make a change that has the minimum amount of side effects, as any side effects could be considered a breaking change. With a project the size of Celery, it's hard to know what folks are using and how... As your own use case proves (I didn't forsee it coming).

Reply:
I agree that its best to avoid breaking changes. Having said that, reducing complexity in a project the size of Celery is something I would also keep in consideration :) Especially when: 1) we don't know at all if there are people that want to opt-out from the DT injection 2) there is an easy workaround, as described above.

I believe you will choose whatever solution is best for the project, and I don't have any strong opinions here (nor I have any time pressure as I have fixed this issue on my side).

Reply:
for v.5.4.x we need to keep the backward compatibility. in future major versions, we can think of improving it and deprecating un needed API.